x-pack/plugins/snapshot_restore/server/routes/api/snapshots.ts

```diff
@@ -55,18 +116,44 @@ export function registerSnapshotsRoutes({
         return handleEsError({ error: e, response: res });
       }
 
+      // if the search is for a repository name with exact match (repository=test)
+      // and that repository doesn't exist, ES request throws an error
+      // that is why we return an empty snapshots array instead of sending an ES request
+      if (
+        searchField === 'repository' &&
+        isSearchingForNonExistentRepository(repositories, searchValue!, searchMatch, searchOperator)
+      ) {
+        return res.ok({
+          body: {
+            snapshots: [],
+            policies,
+            repositories,
+            errors: [],
+            total: 0,
+          },
+        });
+      }
       try {
         // If any of these repositories 504 they will cost the request significant time.
         const { body: fetchedSnapshots } = await clusterClient.asCurrentUser.snapshot.get({
-          repository: '_all',
-          snapshot: '_all',
+          repository:
+            searchField === 'repository'
+              ? convertSearchToWildcard(searchValue!, searchMatch, searchOperator)
```

Comment:
With all of this new behavior determined by the server, we really need to add some API integration tests or functional tests to guard against regressions. I'd love to see these tests also cover all of the different ways `convertSearchToWildcard` translates from the EUI search syntax to ES parameters.